core/src/main/java/com/linecorp/armeria/client/retry/RetryingHttpClient.java

```diff
@@ -120,16 +120,15 @@ protected HttpResponse doExecute(ClientRequestContext ctx, HttpRequest req) thro
     private void doExecute0(ClientRequestContext ctx, HttpRequestDuplicator rootReqDuplicator,
                             DeferredHttpResponse deferredRes) {
         if (!setResponseTimeout(ctx)) {
-            closeOnException(deferredRes, rootReqDuplicator, ResponseTimeoutException.get());
+            closeOnException(ctx, deferredRes, rootReqDuplicator, ResponseTimeoutException.get());
             return;
         }
 
-        final HttpResponse response;
+        HttpResponse response;
         try {
             response = executeDelegate(ctx, rootReqDuplicator.duplicateStream());
         } catch (Exception e) {
-            closeOnException(deferredRes, rootReqDuplicator, e);
-            return;
+            response = HttpResponse.ofFailure(e);
```

Comment:
Just curious. Does it aim to apply the retry strategy when catching an exception from the delegate?

Reply:
You are right. That is the reason.